File: ms-flight-search/src/main/java/dev/patricksilva/model/services/FlightSearchResult.java
package dev.patricksilva.model.services;

import java.util.List;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;

public class FlightSearchResult {
	@Autowired
	private List<Map<String, Object>> flightsOneWay;
    
	@Autowired
    private List<Map<String, Object>> flightsReturn;

	public FlightSearchResult(List<Map<String, Object>> flightsOneWay) {
        this.flightsOneWay = flightsOneWay;
    }
	
    public FlightSearchResult(List<Map<String, Object>> flightsOneWay, List<Map<String, Object>> flightsReturn) {
        this.flightsOneWay = flightsOneWay;
        this.flightsReturn = flightsReturn;
    }

    public List<Map<String, Object>> getFlightsOneWay() {
        return flightsOneWay;
    }

    public List<Map<String, Object>> getFlightsReturn() {
        return flightsReturn;
    }

    public void setFlightsOneWay(List<Map<String, Object>> flightsOneWay) {
		this.flightsOneWay = flightsOneWay;
	}

	public void setFlightsReturn(List<Map<String, Object>> flightsReturn) {
		this.flightsReturn = flightsReturn;
	}
}

File: ms-GO-flight-search/model/services/FlightSearchResult.go
package services

type FlightSearchResult struct {
	FlightsOneWay []map[string]interface{}
	FlightsReturn []map[string]interface{}
}

func NewFlightSearchResult(flightsOneWay, flightsReturn []map[string]interface{}) *FlightSearchResult {
	return &FlightSearchResult{
		FlightsOneWay: flightsOneWay,
		FlightsReturn: flightsReturn,
	}
}

func (result *FlightSearchResult) SetFlightsOneWay(flightsOneWay []map[string]interface{}) {
	result.FlightsOneWay = flightsOneWay
}

func (result *FlightSearchResult) SetFlightsReturn(flightsReturn []map[string]interface{}) {
	result.FlightsReturn = flightsReturn
}

func (result *FlightSearchResult) GetFlightsOneWay() []map[string]interface{} {
	return result.FlightsOneWay
}

func (result *FlightSearchResult) GetFlightsReturn() []map[string]interface{} {
	return result.FlightsReturn
}
